pkg/config/message_broker.go

```diff
@@ -0,0 +1,74 @@
+package config
+
+import (
+	"github.com/spf13/pflag"
+
+	mqttoptions "open-cluster-management.io/api/cloudevents/generic/options/mqtt"
+)
+
+type MessageBrokerConfig struct {
+	SourceID           string                   `json:"source_id"`
+	MessageBrokerType  string                   `json:"message_broker_type"`
+	MQTTOptions        *mqttoptions.MQTTOptions `json:"mqtt"`
+	MQTTBrokerHostFile string                   `json:"mqtt_broker_host_file"`
+	MQTTUserNameFile   string                   `json:"mqtt_username_file"`
+	MQTTPasswordFile   string                   `json:"mqtt_password_file"`
+}
+
+func NewMessageBrokerConfig() *MessageBrokerConfig {
+	mqttOptions := mqttoptions.NewMQTTOptions()
```

Comment:
set all of your config in this file

Reply:
How can we make sure the MQTT options and sourceID are correctly set, if we create CloudEventSourceClient here? `NewMessageBrokerConfig` is called before `AddFlags` .